books/RayTracingInOneWeekend.html

```diff
@@ -1696,16 +1742,53 @@
 </div>
 
 <div class='together'>
-Then update the `ray_color()` function to use the new random direction generator:
+Once we have a random vector in the unit sphere we need to normalize to get a vector _on_ the unit
+sphere
+
+  ![Figure [sphere-vec]: The accepted random vector is normalized to produce a unit vector
+  ](../images/fig-1.11-sphere-unit-vec.jpg)
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    inline vec3 random_unit_vector() {
+        return unit_vector(random_in_unit_sphere());
+    }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [random-unit-vec]: <kbd>[vec3.h]</kbd> Random vector on the unit sphere]
+
+And now that we have a random vector on the surface of the unit sphere, we can determine if it is on
+the correct hemisphere by comparing against the surface normal:
+
+  ![Figure [normal-hor]: The normal vector tells us which hemisphere we need
+  ](../images/fig-1.12-surface-normal.jpg)
+
+We can take the dot product of the normal and our random vector to determine if it's in the correct
+hemisphere: If the dot product is positive then the vector is in the correct hemisphere, if
```

Comment:
"If the dot product is positive, then ... . If the dot product is negative, then ... ."